kubeadm/KubeClusterHelper.psm1

```diff
@@ -240,14 +240,29 @@ function CleanupPolicyList()
 function WaitForNetwork($NetworkName, $waitTimeSeconds = 60)
 {
     $startTime = Get-Date
+    $triedRestart = $false
 
     # Wait till the network is available
     while ($true)
     {
         $timeElapsed = $(Get-Date) - $startTime
         if ($($timeElapsed).TotalSeconds -ge $waitTimeSeconds)
         {
-            throw "Fail to create the network[($NetworkName)] in $waitTimeSeconds seconds"
+            if ($triedRestart)
+            {
+                throw "Fail to create the network[($NetworkName)] in $waitTimeSeconds seconds, again."
+            }
+            
+            # Try Restarting Flanneld and see if that fixed the problem
+            Write-Host "Network [($NetworkName)] was not created within $waitTimeSeconds seconds..."
+            Write-Host "Attempting to turn FlannelD service off and back on again."
+            Stop-Service -Name FlannelD
+            Start-Sleep 5
+            StartFlanneld
```

Comment:
> service sometimes fails to create 'vxlan0' on first start

this is odd. can you provide more details on why it happens exactly?

Reply:
random though: would waiting for N seconds before the flannel service is started also solve this problem?

Reply:
I have a Kubernetes cluster built of VMs that I've been testing on, running Kubernetes 1.16.3.  The  sole master and 5 workers run Ubuntu 18.04.3 and for the last few day's i've been working on add Windows nodes.  I have 4 windows nodes, currently running sac-1909, but I've observed the exact same behavior on ltsb-2019.  

When FlannelD starts, the first time, a vxlan0 HSN network does not get created and the ps script times out waiting.  In every case, on every test VM, simply restarting FlannelD solves this problem.  I've run this altered script on 4 nodes, today, and on each node it timed out, the 1st time,then restarted and succeeded in the -join operation.

I suspect this is actually a FlannelD issue, but this script change works around the issue and allows the join to succeed and the node to work properly.

Reply:
As to waiting longer; that was actually my 1st thought.  I tried going as high as 300 seconds and that did not solve the issue.  I suspect that FlannelD is actually silently failing on the 1st start, but succeeding after that.  In all of my tests, starting FlannelD again was the only solution for me.

Reply:
Ultimately, I realize this is a work-around at best and would probably be called a "hack" by most. A real fix to FlannelD or Windows would be a better solution.  

However, work-arounds are something that we Windows admins are all to familiar with.

Reply:
are there any logs indicating what is the failure?

Reply:
> Ultimately, I realize this is a work-around at best and would probably be called a "hack" by most. A real fix to FlannelD or Windows would be a better solution.

if we can point the flannel maintainers to the the exact failure this can be solved on their side.